features/support/api_helpers.rb

```diff
@@ -333,3 +329,58 @@ def create_user(user_role, supplier_id=nil)
   puts "Email address: #{@user['emailAddress']}"
   @user
 end
+
+def get_or_create_supplier(custom_supplier_data)
+  # This method is used to search for a supplier, and if it does not exist create that supplier. It will return
+  # that supplier and also set it to the global @supplier var.
+  # The possible argument combinations for the getting of a supplier are dependent on the available end points for
+  # suppliers.
+  # Therefore this method supports:
+  # id: being the supplier_id of the supplier (and should not be used with any other args as it's a primary key)
+  # name: name prefix of the supplier. This may result in unpredictable behaviour if the provided name id not unique.
+  # Should the supplier not be found we will attempt a post create with the provided supplier data.
+  if custom_supplier_data["id"] != nil
+    response = call_api(:get, "/suppliers/#{custom_supplier_data['id']}")
+    @supplier = JSON.parse(response.body)["suppliers"]
+  end
+  if custom_supplier_data["name"] != nil
+    response = call_api(:get, '/suppliers', params: {prefix: custom_supplier_data['name']})
+    @supplier = JSON.parse(response.body)['suppliers'][0]
+  end
+  if not @supplier
+    @supplier = create_supplier(custom_supplier_data)
+  end
+  return @supplier
+end
+
+def get_or_create_user(custom_user_data)
+  # This method is used to search for a user, and if it does not exist create that user. It will return that user and
+  # also sets it to the global @user var.
+  # The possible argument combinations for the getting of a user are dependent on the available end points for users.
+  # Therefore this method supports:
+  # id: being the id of the user (and should not be used with any other args as it's a primary key)
+  # email_address: email_address of the account. This can be used independently.
+  # supplier_id: supplier id of the user. May result in unpredictable behaviour if there is more than one user with the
```

Comment:
The description here isn't quite right: the method will only make an API call at all if `email_address` is also specified, it will ignore `supplier_id` otherwise.

Not sure if that should be the desired behaviour though? Getting a user for a particular supplier seems useful, and we probably don't care exactly which email address the user has.

Reply:
Good catch, should make the call regardless of email Done 👍